account_ecotax/models/account_move_line.py

```diff
@@ -0,0 +1,119 @@
+# © 2014-2023 Akretion (http://www.akretion.com)
+#   @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import Command, api, fields, models
+
+
+class AcountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    ecotax_line_ids = fields.One2many(
+        "account.move.line.ecotax",
+        "account_move_line_id",
+        string="Ecotax lines",
+        copy=True,
+    )
+    subtotal_ecotax = fields.Float(
+        string="Ecotax", digits="Ecotax", store=True, compute="_compute_ecotax"
+    )
+    ecotax_amount_unit = fields.Float(
+        digits="Ecotax",
+        string="Ecotax Unit",
+        store=True,
+        compute="_compute_ecotax",
+    )
+
+    @api.depends(
+        "currency_id",
+        "tax_ids",
+        "quantity",
+        "product_id",
+    )
+    def _compute_ecotax(self):
+        for line in self:
+            ecotax_ids = line.tax_ids.filtered(lambda tax: tax.is_ecotax)
+
+            if line.display_type == "tax" or not ecotax_ids:
+                continue
+            if line.display_type == "product" and line.move_id.is_invoice(True):
+                amount_currency = line.price_unit * (1 - line.discount / 100)
+                handle_price_include = True
+                quantity = line.quantity
+            else:
+                amount_currency = line.amount_currency
+                handle_price_include = False
+                quantity = 1
+            compute_all_currency = ecotax_ids.compute_all(
+                amount_currency,
+                currency=line.currency_id,
+                quantity=quantity,
+                product=line.product_id,
+                partner=line.move_id.partner_id or line.partner_id,
+                is_refund=line.is_refund,
+                handle_price_include=handle_price_include,
+                include_caba_tags=line.move_id.always_tax_exigible,
+            )
+            subtotal_ecotax = 0.0
+            for tax in compute_all_currency["taxes"]:
+                subtotal_ecotax += tax["amount"]
+
+            unit = subtotal_ecotax / quantity if quantity else subtotal_ecotax
```

Comment:
```suggestion
            unit = subtotal_ecotax * quantity if quantity else subtotal_ecotax
```



see https://github.com/OCA/l10n-france/blob/04243404949a9dc13bd45a8e8c4675e5332493b1/l10n_fr_ecotaxe/models/account_move_line.py#L27C21-L27C62

Reply:
Unit represent the ecotax amount for 1 product (for reporting purpose). The multiplication by the quantity is set in python code of taxes (see the readMe file).